src/hooks/use-compare-documents.ts

```diff
@@ -0,0 +1,98 @@
+/*!
+ * The contents of this file are subject to the Common Public Attribution License Version 1.0 (the “License”); you may not use this file except in compliance with the License. You may obtain a copy of the License at https://mpapp-public.gitlab.io/manuscripts-frontend/LICENSE. The License is based on the Mozilla Public License Version 1.1 but Sections 14 and 15 have been added to cover use of software over a computer network and provide for limited attribution for the Original Developer. In addition, Exhibit A has been modified to be consistent with Exhibit B.
+ *
+ * Software distributed under the License is distributed on an “AS IS” basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the specific language governing rights and limitations under the License.
+ *
+ * The Original Code is manuscripts-frontend.
+ *
+ * The Original Developer is the Initial Developer. The Initial Developer of the Original Code is Atypon Systems LLC.
+ *
+ * All portions of the code written by Atypon Systems LLC are Copyright (c) 2025 Atypon Systems LLC. All Rights Reserved.
+ */
+
+import { ManuscriptNode } from '@manuscripts/transform'
+import { useEffect, useState } from 'react'
+
+import { compareDocuments } from '../lib/comparison/compare-documents'
+import { ManuscriptSnapshot } from '../lib/doc'
+
+interface UseCompareDocumentsProps {
+  originalId?: string
+  comparisonId?: string
+  getSnapshot: (id: string) => Promise<ManuscriptSnapshot | undefined>
+  originalDoc: ManuscriptNode
+}
+
+interface UseCompareDocumentsReturn {
+  comparedDoc: ManuscriptNode | null
+  isComparingMode: boolean
+  isLoading: boolean
+  error: string | null
+}
+
+export const useCompareDocuments = ({
+  originalId,
+  comparisonId,
+  getSnapshot,
+  originalDoc,
+}: UseCompareDocumentsProps): UseCompareDocumentsReturn => {
```

Comment:
I don't see a reason to specify return value type here. Typescript would derive the type out of the function return value. If that's just to convert Node into ManuscriptNode - you can just typecast that one variable in the return block of the function